### PR TITLE
Tests for by-id core path

### PR DESCRIPTION
**Pull request by Ostrokrzew**
Needs https://github.com/Open-CAS/test-framework/pull/229

### CODE REVIEW

File: test/functional/tests/cache_ops/test_core_add.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev_link = Symlink(core_dev.system_path).get_target()
+        core_dev.create_partitions([Size(400, Unit.MebiByte)] * cores_number)
+
+    with TestRun.step(f"Create symlinks for {core_dev} partitions in {by_id_dir}."):
+        for i, partition in enumerate(core_dev.partitions):
+            create_symlink(f"{symlink_name}-part{i + 1}", by_id_dir, partition.system_path)
+
+    with TestRun.step(f"Find symlinks in {by_id_dir} to {core_dev_link}."):
```
Comment (Deixx) [inline test/functional/tests/cache_ops/test_core_add.py:49]: to ... in ...

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 2]
```diff
@@ -102,3 +110,111 @@ def test_cas_startup(cache_mode, filesystem):
...
+    """
+    with TestRun.step("Clearing dmesg"):
+        TestRun.executor.run_expect_success("dmesg -C")
+
+    with TestRun.step("Prepare partitions for cache and for cores."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(200, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev_link = Symlink(core_dev.system_path).get_target()
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:136]: same as in test_add_core_path_by_id

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 3]
```diff
@@ -102,3 +110,111 @@ def test_cas_startup(cache_mode, filesystem):
...
+
+    with TestRun.step("Compare paths to cores."):
+        paths_from_config = InitConfig.load_config_file().core_config_lines
+        for i in range(cores_number):
+            if paths_from_config[i].core_path != selected_links[i].full_path:
+                TestRun.LOGGER.error(
+                    f"Paths are different and can cause problems!\n"
+                    f"Path passed as an argument to add core: {selected_links[i].full_path}\n"
+                    f"Path currently used in core addition: {paths_from_config[i].core_path}"
+                )
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:182]: Seems unnecessary - basically checks if we read what we wrote

File: test/functional/api/cas/init_config.py [hunk 4]
```diff
@@ -57,6 +58,50 @@ def create_default_init_config(cls):
         cas_version = casadm_parser.get_casadm_version()
         fs_utils.write_file(opencas_conf_path, f"version={cas_version.base}")
 
+    @classmethod
+    def load_config_file(cls):
```
Comment (Deixx) [inline test/functional/api/cas/init_config.py:62]: Considering comment for test_startup_init_config.py lines 174+, this method might be unnecessary

**Review by Deixx — CHANGES_REQUESTED** [review]

**Review by katlapinka — CHANGES_REQUESTED** [review]

**Comment by karolinavelkaja** [comment]
Closing since Sławek is no longer in our team and this PR needed few changes. 
New review: https://github.com/Open-CAS/open-cas-linux/pull/931

**Pull request closed without merging**